services/publisher.py

```
@@ -0,0 +1,175 @@
+    def _r2_file_exists(self, key: str) -> bool:
+        """Check if file exists in R2 bucket using Cloudflare REST API."""
+        try:
+            url = f"https://api.cloudflare.com/client/v4/accounts/{self.cloudflare_account_id}/storage/buckets/{self.r2_bucket}/objects/{key}"
+            
+            with httpx.Client(timeout=httpx.Timeout(connect=10.0, read=30.0)) as client:
+                response = client.head(
+                    url,
+                    headers={"Authorization": f"Bearer {self.cloudflare_api_token}"}
+                )
+                return response.status_code == 200
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)